controllers/storagecluster/exporter.go

```diff
@@ -1114,3 +1115,16 @@ func (r *StorageClusterReconciler) deleteMetricsExporterCephClient(namespace str
 
 	return nil
 }
+
+// shouldUseHostNetworking checks if the StorageCluster is configured to run on non-default host network
+// this is validated by checking if the AddressRanges.Public field is set in the Network spec
+// since pod network cannot always communicate with non-default host network metrics exporter needs to run on host net
+func shouldUseHostNetworking(instance *ocsv1.StorageCluster) bool {
```

Comment:
this part should be removed when the following code is in
https://github.com/red-hat-storage/ocs-operator/pull/3423/files#diff-4160c186e4e6b63623e530892b0aa7f6385e7f60f45a5f3da944492ea6211787R969